src/main/scala/com/metabolic/data/core/services/spark/writer/file/IcebergWriter.scala

```diff
@@ -0,0 +1,82 @@
+package com.metabolic.data.core.services.spark.writer.file
+
+import com.metabolic.data.core.services.spark.writer.DataframeUnifiedWriter
+import com.metabolic.data.mapper.domain.io.WriteMode
+import com.metabolic.data.mapper.domain.io.WriteMode.WriteMode
+import org.apache.spark.sql.catalyst.dsl.expressions.StringToAttributeConversionHelper
+import org.apache.spark.sql.functions.days
+import org.apache.spark.sql.streaming.{StreamingQuery, Trigger}
+import org.apache.spark.sql.{DataFrame, SparkSession}
+
+import java.util.concurrent.TimeUnit
+
+class IcebergWriter(
+                     val fqn: String,
+                     val writeMode: WriteMode,
+                     val checkpointLocation: String)
+                   (implicit  val spark: SparkSession)
+  extends DataframeUnifiedWriter {
+
+  override val output_identifier: String = fqn
+
+  override def writeBatch(df: DataFrame): Unit = {
+
+    writeMode match {
+      case WriteMode.Append => df
+        .write
+        .format("iceberg")
+        .mode("append")
+        .saveAsTable(output_identifier)
+
+      case WriteMode.Overwrite => df
+        .write
+        .format("iceberg")
+        .mode("overwrite")
+        .saveAsTable(output_identifier)
+
+      case WriteMode.Upsert =>
+        throw new NotImplementedError("Batch Upsert is not supported in Iceberg yet")
```

Comment:
How would they do incremental overwrites?

Reply:
We definitely need to add the MERGE INTO option

Reply:
Comming back to this, incremental overwrite should be done with the op replace, which replaces the table with the information of the dataframe regardless of the schema. 

I have added an implementation of the op overwrite which uses https://iceberg.apache.org/docs/1.6.0/spark-writes/#overwriting-data . This requires the new data to maintain the same schema,